kcfinder/integration/civicrm.php

```diff
@@ -36,13 +36,32 @@
  *
  */
 
+
 function checkAuthentication() {
   static $authenticated;
   if ( !isset( $authenticated ) ) {
-    $current_cwd   = getcwd();
-    $civicrm_root  = dirname(dirname(getcwd()));
     $authenticated = false;
-    require_once "{$civicrm_root}/civicrm.config.php";
+
+    // used to chdir at the end of this function - not sure if necessary?
+    $current_cwd = getcwd();
```

Comment:
Oh, that is funny.

+1 for what you've done here - keep for now, but flag it to see as a curiosity.

* FWIW, it is plausible that this is needed. The effect of loading `civicrm.config.php` is to bootstrap some mix of Civi+UF (BD/D7/D10/J/WP) -- and it's quite possible that *some* combination of Civi+UF has the side-effect of changing CWD (*to match the UF's project-root or HTTPD web-root*).

* Our E2E test-coverage over this bit is non-existent. (The more we change this file, the more we have to think about QA work.)